Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1158,16 +1169,25 @@ void CompareWorkspaces::doLeanElasticPeaksComparison(const LeanElasticPeaksWorks
           s1 += (q1[i] - q2[i]) * (q1[i] - q2[i]);
         }
         s1 = std::sqrt(s1);
+        if (isRelErr) {
+          // divide diff by avg |Q| and then compare to 0 using absolute tol
+          s1 /= 0.5 * (q1.norm() + q2.norm());
+        }
       } else {
         g_log.information() << "Column " << name << " is not compared\n";
       }
-      if (std::fabs(s1 - s2) > tolerance) {
+      bool mismatch = false;
+      if (isRelErr && relErr(s1, s2, tolerance) && name != "QLab" && name != "QSample") {
```

Comment:
I think this line probably also needs changing (similar to other comment). The call to relErr needs to be inside the first part of the if...else statement so that the code doesn't perform the absolute check when isRelErr=true